src/firebase/saga.ts

```diff
@@ -36,16 +37,25 @@ const EXCHANGE_RATES = 'exchangeRates'
 const VALUE_CHANGE_HOOK = 'value'
 const FIREBASE_CONNECT_RETRIES = 3
 
-let firebaseAlreadyAuthorized = false
-export function* waitForFirebaseAuth() {
-  if (firebaseAlreadyAuthorized) {
+const actionSeen: Record<string, boolean> = {}
+
+export function* waitForAction(action: Actions | HomeActions) {
```

Comment:
I worry that `takeOnce` over promises, since the in-memory cache of actions seen will be reset whenever the app restarts. I went about renaming and documenting this, but people don't always read the fine print when something is named a certain way

Reply:
maybe `takeWithInMemoryCache` would be transparent enough? bit of a mouthful but I'm fine with it

Reply:
good point. `takeOnceInMemory`? 

Also fine with a small comment to ward people off from just using without understanding.

Reply:
i removed the changes here so hitting resolve on this one